test-framework/junit5/src/main/java/org/apache/camel/quarkus/test/CallbackUtil.java

```diff
@@ -46,15 +46,22 @@ static Optional<TestInstance.Lifecycle> getLifecycle(CamelQuarkusTestSupport tes
         return Optional.empty();
     }
 
-    static void resetContext(CamelQuarkusTestSupport testInstance) {
+    static void resetContext(CamelQuarkusTestSupport testInstance, boolean keepOriginalRoutes) {
 
-        //if routeBuilder (from the test) was used, all routes has to be stopped and removed
+        //if routeBuilder (from the test) was used, all routes from that builder has to be stopped and removed
         //because routes will be created again (in case of TestInstance.Lifecycle.PER_CLASS, this method is not executed)
-        if (testInstance.isUseRouteBuilder()) {
-            try {
-                testInstance.context().getRouteController().stopAllRoutes();
-                testInstance.context().getRouteController().removeAllRoutes();
+        if (testInstance.isUseRouteBuilder() && testInstance.createdRoutes != null) {
 
+            try {
+                if (keepOriginalRoutes) {
```

Comment:
Could you please details what `keepOriginal` means in this context ?

It's like we keep the state of route ? Or we want a fresh route ?

Reply:
I'll add comment here. that makes sense, it didn't occur to me that it is my own definition of routes.

Reply:
I removed this if. It should not be there. There is a following ticket, which I'm starting to work on, which takes care of the lifecycle with Lifecycle.PER_CLASS - https://github.com/apache/camel-quarkus/issues/4569